ansible_base/lib/abstract_models/common.py

```diff
@@ -77,39 +75,13 @@ class Meta:
         help_text="The user who last modified this resource",
     )
 
-    def _attributable_user(self, non_existent_user_fatal=True):
-        # returns:
-        #    The logged in user object
-        #    The system user object
-        #    None if there is no user and the system user is not defined
-        #    AnonymousUser if an anonymous request came in
-
-        user = get_current_user()
-
-        # We might be AnonymousUser but there could be conditions where something might need to save even if we are Anonymous.
-        convert_anonymous_user_to_system = False
-        if user and user.is_anonymous:
-            # There is a special case where, when creating an user from an authenticator we will be anonymous user until the user is actually created
-            if not self.pk and isinstance(self, AbstractUser):
-                convert_anonymous_user_to_system = True
-
-        if user is None or convert_anonymous_user_to_system:
-            # If no user is logged in, we try attributing the action to the system user
-            # If there is no system username defined, we just leave the user as None
-            system_username = get_setting('SYSTEM_USERNAME')
-            if system_username is not None:
-                try:
-                    user = get_user_model().objects.get(username=system_username)
-                except get_user_model().DoesNotExist:
-                    if non_existent_user_fatal:
-                        logger.warning(f"SYSTEM_USERNAME is set to {system_username} but no user with that username exists.")
-                    user = None
-
-        return user
-
     def _check_user(self, user, non_existent_user_fatal=True):
-        # Its possible for _attributable_user to return None or anonymous user in which case we don't want to save
-        if (not user or user.is_anonymous) and non_existent_user_fatal and get_setting('SYSTEM_USERNAME'):
+        # user might be:
+        #     The logged in user object; save is allowed
+        #     The system user object; save is allowed
+        #     None; Do not allow save (unless non_existent_user_fatal is False)
+        #     AnonymousUser if an anonymous request came in; Do not allow save (unless non_existent_user_fatal)
+        if (not user or user.is_anonymous) and non_existent_user_fatal:
```

Comment:
I worry about this method in general.  Django ties transaction management to whether or not an exception has been raised.  I can't remember off the top of my head how it goes, but I can easily imagine this rippling out to rollback more objects than intended.

Also looming large in my mind is the question of whether or not non-web actions (shell, management commands, internal processes, etc) should touch these fields for any object.  I kind of lean towards 'no'.

Reply:
If we are in a transaction saving multiple things and one fails because of this I would think the transaction should roll back. Do you feel differently? 

I think non-web actions should still modify these things as they are kind of like "audit trails" in my mind. A management command or internal process I think should do a `with impersonate(system_user)` if applicable or somehow otherwise figure out how to authenticate who the user is (admin presumably if they are running a management command) and update the fields with those fields. 

But I'm open to going with a group concensious on this point.